### PR TITLE
5_types revised

### PR DESCRIPTION
**Pull request by grosu**
I have tried to thorought revise 5_types, but I had to give up.  It requires a few days of work itself before it is in a good shape.  It was in a similarly bad shape before, when we released 3.6.  Under the circumstances, I think we should go ahead and merge this bit of cleanup now, and then, after release, I'll take a few days to dedicate to improving this tutorial on types.  We should also publish a paper about our approach, as it is not trivial and it quite interesting.  If any of you is interested in joining, let me know.

### CODE REVIEW

**Comment by cos** [comment]
checkstyle failed

```
/var/lib/jenkins/jobs/k-framework-pull-request/workspace/k-distribution/tutorial/1_k/5_types/lesson_9/README.md:1: Line does not match expected header line of '^<!-- Copyright \(c\) (\d\d\d\d-)?2016 K Team. All Rights Reserved. -->$'.
/var/lib/jenkins/jobs/k-framework-pull-request/workspace/k-distribution/tutorial/1_k/5_types/lesson_5/README.md:1: Line does not match expected header line of '^<!-- Copyright \(c\) (\d\d\d\d-)?2016 K Team. All Rights Reserved. -->$'.
/var/lib/jenkins/jobs/k-framework-pull-request/workspace/k-distribution/tutorial/1_k/5_types/lesson_6/README.md:1: Line does not match expected header line of '^<!-- Copyright \(c\) (\d\d\d\d-)?2016 K Team. All Rights Reserved. -->$'.
```

File: k-distribution/tutorial/1_k/5_types/lesson_3/README.md [hunk 1]
```diff
@@ -28,8 +30,8 @@ type lookup rule:
 
 The type environment is populated by the semantic rule of `lambda`:
 
-    rule <k> lambda X : T . E => (T -> E) ~> tenv(Rho) ...</k>
-         <tenv> Rho => Rho[T/X] </tenv>
+    rule <k> lambda X : T . E => (T -> E) ~> Rho ...</k>
+         <tenv> Rho => Rho[X <- T] </tenv>
```
Comment (cos) [inline k-distribution/tutorial/1_k/5_types/lesson_3/README.md:34]: just curious, why the change of notation?

**Comment by cos** [comment]
Jenkins: test this please

File: k-distribution/tutorial/1_k/5_types/lesson_3/README.md (hunk 1 above)
Comment (grosu) [inline k-distribution/tutorial/1_k/5_types/lesson_3/README.md:34]: which one, `tenv` or the map update?

Comment (cos) [inline k-distribution/tutorial/1_k/5_types/lesson_3/README.md:34]: the map update; but I guess it looks better this way, so that it doesn't get confused with the more general substitution.

Comment (grosu) [inline k-distribution/tutorial/1_k/5_types/lesson_3/README.md:34]: yes, that was the original reason.  note that all definitions use it: lambda, imp, simple, kool, etc.  it was not changed here, but a long time ago.  i didn't update the tutorial for a long time, that's the real problem
